CMakeLists.txt

```diff
@@ -0,0 +1,403 @@
+# User note:
+# This aims to work almost identical to libtool-based compilation pipeline.
+# At project root, run:
+# $ mkdir build && cd build/
+# $ cmake -DMAKE_INSTALL_PREFIX=$VIRTUAL_ENV ..
+
+# Minimum required CMake version
+cmake_minimum_required(VERSION 3.10)
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+# Project/Package name and version
+project(fplll VERSION 5.4.5)
+set(FPLLL_MAJOR_VERSION ${fplll_VERSION_MAJOR})
+set(FPLLL_MINOR_VERSION ${fplll_VERSION_MINOR})
+set(FPLLL_MICRO_VERSION ${fplll_VERSION_PATCH})
+set(FPLLL_VERSION ${fplll_VERSION})
+
+if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+  if(CMAKE_CXX_COMPILER_VERSION MATCHES "^14.")
+    message(STATUS "You're using ${CMAKE_CXX_COMPILER_ID} ${CMAKE_CXX_COMPILER_VERSION}")
+  else()
+    message(WARNING "!! We recommend that you use clang-14 for this bootcamp. You're using ${CMAKE_CXX_COMPILER_ID} ${CMAKE_CXX_COMPILER_VERSION}, a different version.")
+  endif()
+elseif(CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
+  message(STATUS "You're using ${CMAKE_CXX_COMPILER_ID} ${CMAKE_CXX_COMPILER_VERSION}")
+else()
+  message(WARNING "!! We recommend that you use clang-14 for this bootcamp. You're using ${CMAKE_CXX_COMPILER_ID} ${CMAKE_CXX_COMPILER_VERSION}, which is not clang.")
+endif()
+
+# standard library use c++11
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
+set(CMAKE_SYSROOT $ENV{SDKROOT})
+
+# Find system dependencies
+# GMP
+find_path(GMP_INCLUDE_DIR gmp.h PATHS /opt/homebrew/include /opt/homebrew/opt/gmp/include)
```

Comment:
This assumes an M1+ mac (see [this](https://github.com/fplll/fplll/commit/e734105077ab1a63722a1971338d38ce93274565)). Could we make a best-faith effort to try to find these headers i.e. could we not assume that they're installed in a particular fixed location?

Reply:
this is very true. agree that we should improve